From 01d5994dcdf783ce9037695eb547a6456c57cf78 Mon Sep 17 00:00:00 2001
From: ming_qian <ming.qian@nxp.com>
Date: Mon, 8 Jul 2019 13:33:42 +0800
Subject: [PATCH 14/52] MLK-22038: don't clear instance before reset firmware

reset firmware may be called after snapshot,
in this case, we shouldn't clear instance,
otherwise, the stream will hang

Signed-off-by: ming_qian <ming.qian@nxp.com>
Reviewed-by: Shijie Qin <shijie.qin@nxp.com>
---
 drivers/mxc/vpu_malone/vpu_b0.c | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/drivers/mxc/vpu_malone/vpu_b0.c b/drivers/mxc/vpu_malone/vpu_b0.c
index 3b4a8edc6b0a..4e279eb019ed 100644
--- a/drivers/mxc/vpu_malone/vpu_b0.c
+++ b/drivers/mxc/vpu_malone/vpu_b0.c
@@ -5708,17 +5708,6 @@ static int swreset_vpu_firmware(struct vpu_dev *dev, u_int32 idx)
 	vpu_dbg(LVL_WARN, "SWRESET: swreset_vpu_firmware\n");
 	dev->firmware_started = false;
 
-	ctx->firmware_stopped = true;
-	ctx->start_flag = true;
-	ctx->b_firstseq = true;
-	ctx->wait_rst_done = false;
-	down(&ctx->q_data[V4L2_DST].drv_q_lock);
-	respond_req_frame(ctx, &ctx->q_data[V4L2_DST], true);
-	clear_queue(&ctx->q_data[V4L2_DST]);
-	reset_mbi_dcp_count(ctx);
-	up(&ctx->q_data[V4L2_DST].drv_q_lock);
-	memset(ctx->pSeqinfo, 0, sizeof(MediaIPFW_Video_SeqInfo));
-	vpu_dec_cleanup_cmd(ctx);
 	v4l2_vpu_send_cmd(ctx, 0, VID_API_CMD_FIRM_RESET, 0, NULL);
 
 	reinit_completion(&dev->start_cmp);
-- 
2.17.1

